scipy/stats/unuran/unuran_wrapper.pyx.templ

```diff
@@ -278,6 +278,43 @@ cdef void _pack_dist(unur_distr *distr, dict callbacks):
             unur_distr_discr_set_cdf(distr, discr_cdf_thunk)
 
 
+def _validate_domain(domain):
+    if domain is not None:
+        # UNU.RAN doesn't recognize nans in the probability vector
+        # and throws an "unknown error". Hence, check for nans ourselves
+        if np.isnan(domain).any():
+            raise ValueError("`domain` must contain only non-nan values.")
+        # Length of the domain must be exactly 2.
+        if len(domain) != 2:
+            raise ValueError("`domain` must be a length 2 tuple.")
+        # Throw an error here if it can't converted into a tuple.
+        domain = tuple(domain)
+    return domain
+
+
+cdef double[::1] _validate_pv(pv) except *:
+    cdef double[::1] pv_view = None
+    if pv is not None:
+        # Make sure the PV is a contiguous array of doubles.
+        pv = pv_view = np.ascontiguousarray(pv, dtype=np.float64)
+        # Empty arrays not allowed.
+        if pv.size == 0:
+            raise ValueError("`pv` must have at least one element.")
+        # NaNs not recognized by UNU.RAN so throw an error here only.
+        if np.isnan(pv).any():
+            raise ValueError("`pv` must contain only non-nan values.")
+        # inf values not recognized by UNU.RAN so throw an error here only.
+        if not np.isfinite(pv).all():
```

Comment:
I think we can simplify the checks slightly to cover NaN that are also detected by `np.isfinite` if we extend the error message to: pv must contain only finite / non-nan values

Reply:
Done.